sui/src/sui_json.rs

```diff
@@ -327,19 +286,50 @@ fn resolve_object_args(
 /// This is because we have special types which we need to specify in other formats
 pub fn resolve_move_function_args(
     package: &Object,
-    module: Identifier,
+    module_ident: Identifier,
     function: Identifier,
     combined_args_json: Vec<SuiJsonValue>,
 ) -> Result<(Vec<ObjectID>, Vec<Vec<u8>>), anyhow::Error> {
     // Extract the expected function signature
-    let function_signature = package
+    let module = package
         .data
         .try_as_package()
         .ok_or_else(|| anyhow!("Cannot get package from object"))?
-        .get_function_signature(&module, &function)?;
+        .deserialize_module(&module_ident)?;
+    let function_str = function.as_ident_str();
+    let fdef = module
+        .function_defs
+        .iter()
+        .find(|fdef| {
+            module.identifier_at(module.function_handle_at(fdef.function).name) == function_str
+        })
+        .ok_or_else(|| {
+            anyhow!(
+                "Could not resolve function {} in module {}",
+                function,
+                module_ident
+            )
+        })?;
+    let function_signature = module.function_handle_at(fdef.function);
+    let parameters = &module.signature_at(function_signature.parameters).0;
```

Comment:
What the reason to stop calling `get_function_signature`? This is essentially what the package cache is trying to cache: to cache function signatures (as they never change, and we shouldn't do this again and again)

Reply:
That function doesn't exist anymore. I imagine the caching doesn't matter for `sui_json`?

On the adapter side of things, the caching is now managed by the VM (which was already caching the modules)

Reply:
This is not the same cache that Ade mentioned on the `sui_json` side. @gdanezis wrote this cache.

This cache is to speed up `resolve_object_args` across executions. Say two transactions are resolving object args on the same function, the cache allows it to skip deserialization the second type the same package/function is encountered, and can return the signature directly.